### DISCUSSION TITLE
Incorprating changes back to Power Platform

### DISCUSSION THREAD

**Discussion by fadyanwar** [microsoft/Power-Fx#343] [Q&A] [answered]
Hello, hope all is well. I was wondering if there are any future plans to allow developers to incorporate their changes to Power Fx into their solutions on Power Platform. i.e. allowing developers to upload their Power Fx newly implemented helper functions in a plugin like fashion to power platform so they can be used in Power Automate or Canvas Apps. Thanks!

Fady

**Comment by MikeStall** [comment, marked as answer]
Just curious - do you have some particular functions that you wanted here?

If it's a "standard" function, we try to roll it into the core implementation. 

If it's truly a "custom" function,  then we are looking at various extensibility options. 
One challenge is that Power Apps runs Fx as JavaScript (not C#). 

1. We're looking at allowing running power fx in a dataverse custom API. 
2. Or for cases of already running power fx in the sandbox, we may have some way to allow you to provide your function implementations.

**Reply by fadyanwar** [reply]
I was asking generally but mostly extending PowerFx on Power Platform would be useful for functions that can't be standardized yet can be deployed in solutions for specific verticals. Option 2 sounds more like it, is this on the roadmap?

**Reply by MikeStall** [reply]
It's on the roadmap ...  but at least several months away...

**Comment by fadyanwar** [comment]
Hi @MikeStall just following up on this to see if there have been any updates. Thanks.

> It's on the roadmap ... but at least several months away...
